rsc.js

```diff
@@ -60,30 +47,26 @@ module.exports.createFactory ||= function unsupportedCreateFactory() {
   };
 };
 
-// Trick cjs-module-lexer into adding named exports for these names.
-// (they will appear in `.default` as well.)
-// NOTE: these names need to appear verbatim, no variables or anything!
-if (0) {
-  const DUMMY = /** @type {any} */ (1);
-  // missingFunctions
-  module.exports.createContext = DUMMY;
-  module.exports.createFactory = DUMMY;
-  module.exports.act = DUMMY;
-  module.exports.unstable_act = DUMMY;
-  module.exports.unstable_useCacheRefresh = DUMMY;
-  module.exports.useContext = DUMMY;
-  module.exports.useDeferredValue = DUMMY;
-  module.exports.useEffect = DUMMY;
-  module.exports.useImperativeHandle = DUMMY;
-  module.exports.useInsertionEffect = DUMMY;
-  module.exports.useLayoutEffect = DUMMY;
-  module.exports.useReducer = DUMMY;
-  module.exports.useRef = DUMMY;
-  module.exports.useState = DUMMY;
-  module.exports.useSyncExternalStore = DUMMY;
-  module.exports.useTransition = DUMMY;
-  module.exports.useOptimistic = DUMMY;
-  // missingClasses
-  module.exports.Component = DUMMY;
-  module.exports.PureComponent = DUMMY;
+// actually overwrite everything with the original module
+Object.assign(module.exports, require("react"));
```

Comment:
Here we add the react exports - and if we accidentally specified one of those above here, it will be overwritten.

Reply:
ah, i missed this bit, yeah. so this'd overwrite our stubs if a real impl is present. makes sense! in this case i think we can also remove the `||=` from createContext/createFactory for consistency?

Reply:
Good point, I added a bit more of a comment :)